sopel/irc/__init__.py

```diff
@@ -830,18 +835,22 @@ def say(
                     time.sleep(sleep_time)
 
             # Loop detection
-            messages = [m[1] for m in recipient_stack['messages'][-8:]]
+            if antiloop_threshold > 0 and elapsed < antiloop_window:
+                messages = [m[1] for m in recipient_stack['messages'][-10:]]
 
-            # If what we're about to send repeated at least 5 times in the last
-            # two minutes, replace it with '...'
-            if messages.count(text) >= 5 and elapsed < 120:
-                text = '...'
-                if messages.count('...') >= 3:
-                    # If we've already said '...' 3 times, discard message
-                    return
+                # If what we're about to send repeated at least N times
+                # in the anti-looping window, replace it
+                if messages.count(text) >= antiloop_threshold:
+                    text = antiloop_repeat_text
+                    if messages.count(text) >= antiloop_silent_after:
+                        # If we've already said that N times, discard message
+                        return
 
             self.backend.send_privmsg(recipient, text)
-            recipient_stack['flood_left'] = max(0, recipient_stack['flood_left'] - 1)
+
+            # update recipient meta-data
+            flood_left = recipient_stack['flood_left'] - 1
+            recipient_stack['flood_left'] = max(0, flood_left)
             recipient_stack['messages'].append((time.time(), safe(text)))
             recipient_stack['messages'] = recipient_stack['messages'][-10:]
```

Comment:
This actually annoys me, upon looking at the whole file in context. I know a "stack" is different from a FIFO queue, but that's what really should be used for the `messages` buffer (specifically, `collections.deque(maxlen=10)`). Maybe I'll take this branch as a base and try to convert it, instead of making Yet Another Tracking Issue.